cmd/posix.go

```diff
@@ -916,7 +916,10 @@ func deleteFile(basePath, deletePath string) error {
 	}
 
 	// Recursively go down the next path and delete again.
-	return deleteFile(basePath, slashpath.Dir(deletePath))
+	// Errors for parent directories shouldn't trickle down.
+	deleteFile(basePath, slashpath.Dir(deletePath))
```

Comment:
The problem with this is if we fail to delete the file for some reason we never return error.

A better fix is to call this function `deleteDir()` which never returns error like how we do here. But delete the file instead in `DeleteFile()`